ID: 7194
Title: Table engine 'dictionary' raise error on all queries in CH 19.15.2.2
Description:
After upgrade clickhouse to 19.15.2.2 all queries to tables with engine 'dictionary' crashed with error:

```sql
select * from dic_clients
```
```
SQL Error [1002]: ClickHouse exception, message: std::exception. Code: 1001, type: std::bad_variant_access, e.what() = Unexpected index, version = 19.15.2.2 (official build)
, host: localhost, port: 8123;
```

Direct access to dictionary work fine

```sql
select dictGetString('clients','name', toUInt64(27019))
```

Table create DDL:
```sql
CREATE TABLE IF NOT EXISTS `dic_clients` (
    `_id`               UInt64,
    `name`              String
) ENGINE = Dictionary(`clients`);
```

Dictionary config:
```xml
<dictionary>
        <name>clients</name>
        <source>
            <mongodb>
                <host>127.0.0.1</host>
                <port>27017</port>
                <user/>
                <password/>
                <db>clickhouse</db>
                <collection>dic.clients</collection>
            </mongodb>
        </source>
        <layout>
            <hashed/>
        </layout>
        <structure>
            <id>
                <name>_id</name>
            </id>
            <attribute>
                <name>name</name>
                <type>String</type>
                <null_value>&lt;undefine&gt;</null_value>
            </attribute>
        </structure>
        <lifetime>1800</lifetime>
    </dictionary>
```